560. Subarray Sum Equals K/SubarraySumEqualsKStep3.java

```diff
@@ -0,0 +1,17 @@
+public class SubarraySumEqualsKStep3 {
+    // 3min
+    public int subarraySum(int[] nums, int k) {
+        int equalKCount = 0;
+        HashMap<Integer, Integer> prefixSumToCount = new HashMap<>();
+        prefixSumToCount.put(0, 1);
+        int prefixSum = 0;
+        for (int index = 0; index < nums.length; index++) {
+            prefixSum += nums[index];
+            if (prefixSumToCount.containsKey(prefixSum - k)) {
+                equalKCount += prefixSumToCount.get(prefixSum - k);
+            }

```

Comment:
下の行は `getOrDefault` を使っているので、こちらも揃えて良さそうです

```suggestion
            equalKCount += prefixSumToCount.getOrDefault(prefixSum - k, 0);
```

Reply:
ありがとうございます。
確かに、使うんだったら揃えればいいじゃん、となりますね。

コードを読み直してみたのですが、`getOrDefault` を使うと読むのが難しく感じました。なので、いっそのこと使わずに書くのが自分にとって分かりやすそうです。
``` Java
for (int index = 0; index < nums.length; index++) {
    prefixSum += nums[index];
    if (prefixSumToCount.containsKey(prefixSum - k)) {
        equalKCount += prefixSumToCount.get(prefixSum - k);
    }
    int newCount = 1;
    if (prefixSumToCount.containsKey(prefixSum)) {
        newCount += prefixSumToCount.get(prefixSum);
    } 
    prefixSumToCount.put(prefixSum, newCount);
}
```